File: ruby/binary/binary.rb
class Binary

  attr_reader :num

  def initialize(num)
    @num = sanitize(num)
  end

  def to_decimal
    num.map.with_index do |x, index|
      x.to_i * (2**index) 
    end.reduce(:+)
  end

  private

    def sanitize(num)
      num.everse.chars
    end
end

File: javascript/binary/binary.js
'use strict';

class Binary {
  constructor(numString) {
    this.num = numString.split("")
                        .reverse()
                        .map(n => parseInt(n));
  }

  toDecimal() {
    let binary = this.num.reduce( (prev, curr, idx) => 
      prev + (curr * Math.pow(2, idx))
    );

    return binary || 0;
  }
}

module.exports = Binary;
